src/libOpenImageIO/color_ocio.cpp

```diff
@@ -322,9 +322,9 @@ ColorConfig::reset(string_view filename)
     m_impl.reset(new ColorConfig::Impl);
 #ifdef USE_OCIO
     OCIO::SetLoggingLevel(OCIO::LOGGING_LEVEL_NONE);
-    if (!ocio_current_config)
-        ocio_current_config = OCIO::GetCurrentConfig();
     try {
+        if (!ocio_current_config)
+            ocio_current_config = OCIO::GetCurrentConfig();
         if (filename.empty()) {
             getImpl()->config_  = ocio_current_config;
             string_view ocioenv = Sysutil::getenv("OCIO");
```

Comment:
The fix is the right one to avoid the unexpected OCIO exception.

[Comment] The code could end up loading two different configs if OCIO exists and filename is not empty. And the `OCIO` env. variable must have the priority. So, the code could be improved:
1. **if** the `OCIO` env. variable exists the code calls `OCIO::GetCurrentConfig()`
2. **else if** filename not empty the code calls `config = OCIO::Config::CreateFromFile()`, and `OCIO::SetCurrentConfig(config)`
3. **else** `config = OCIO::Config::CreateRaw()`, and `OCIO::SetCurrentConfig(config)`

It now means that the OIIO code only needs to call `OCIO::GetCurrentConfig()` to access the current config instance.

Reply:
At line 338 and 341, the code could do `getImpl()->config_ = OCIO::Config::CreateRaw()` or `config = OCIO::Config::CreateRaw(); OCIO::SetCurrentConfig(config);` if a config instance is mandatory

Reply:
I see, thanks. So the issue is that GetCurrentConfig() will read a config if $OCIO is set? The filename is definitely intended to override $OCIO (potentially per color conversion operation). Let me take a stab at this and see if I can fix the logic so that it won't read two configs.